aws_topology/stackstate_checks/aws_topology/flowlogs.py

```diff
@@ -0,0 +1,340 @@
+
+        self._delete_files(client, bucket_name, to_delete)
+        number_of_files = len(files_to_handle)
+        if number_of_files > 0:
+            connections = self.process_files(client, bucket_name, files_to_handle, nwinterfaces)
+            count = self.process_connections(connections)
+            self.log.info("Found {} S3 objects with {} connections".format(number_of_files, count))
+
+    def _delete_files(self, client, bucket_name, files):
+        for i in range(0, len(files), self.MAX_S3_DELETES):
+            try:
+                self.log.info(
+                    "Deleting {} files from S3 bucket {}".format(len(files[i : i + self.MAX_S3_DELETES]), bucket_name)
+                )
+                client.delete_objects(
+                    Bucket=bucket_name,
+                    Delete={"Objects": files[i : i + self.MAX_S3_DELETES], "Quiet": True},
+                )
+            except Exception as e:
+                self.log.exception(e)
+                self.agent.warning("CloudtrailCollector: Deleting s3 files failed")
+
+    def process_record(self, connections, log, nwitf):
+        src_ip = log["srcaddr"]
+        dst_ip = log["dstaddr"]
+        src_port = log["srcport"]
+        dst_port = log["dstport"]
+        start = int(log.get("start", "0"))
+        end = int(log.get("end", "0"))
+        protocol = log.get("protocol", "unknown_protocol")
+        bytes_transfered = int(log.get("bytes", "0"))
+        dir = "unknown"
+        private = None
+        family = "v4" if ip_version(src_ip) == 4 else "v6"
+        nwitf_update = {}
+        id = ""
+        reverse = None
+        if src_ip in nwitf.addresses:
+            laddr = "{}:{}".format(src_ip, src_port)
+            raddr = "{}:{}".format(dst_ip, dst_port)
+            private = is_private(dst_ip)
+            dir = "out"
+            nwitf_update = {src_ip: nwitf}
+            id, reverse = connection_identifier(nwitf.VpcId, src_ip, dst_ip)
+        elif dst_ip in nwitf.addresses:
+            laddr = "{}:{}".format(dst_ip, dst_port)
+            raddr = "{}:{}".format(src_ip, src_port)
+            private = is_private(src_ip)
+            dir = "in"
+            nwitf_update = {dst_ip: nwitf}
+            id, reverse = connection_identifier(nwitf.VpcId, dst_ip, src_ip)
+        else:
+            self.log.warning("Could not determine traffic direction src={} dst={}".format(src_ip, dst_ip))
+
+        if private and dir != "unknown":  # currently only supporting private traffic
+            logline = "{}:{} <-> {}:{} bytes={}".format(src_ip, src_port, dst_ip, dst_port, bytes_transfered)
+            conn = connections.get(id, None)
+            if conn:
+                conn.add_traffic(start, end, protocol, dir == "in", bytes_transfered, nwitf_update, reverse, logline)
+            else:
+                connections[id] = Connection(
+                    nwitf.VpcId,
+                    family,
+                    laddr,
```

Comment:
this will fail as `laddr` and `raddr` can be referenced before assigment. line 202 `if elif else` statement resolves to else.